examples/src/sysio-write.c

```diff
@@ -345,16 +345,32 @@ int main(int argc, char** argv)
 
     MPI_Barrier(MPI_COMM_WORLD);
 
+    /* have rank 0 check the expected file size matches actual file size */
     if (rank == 0) {
+        /* compute expected size of file after all procs have written,
+         * each process writes nblocks in groups of nchunks each of
+         * which is chunksize bytes */
+        uint64_t nchunks = blocksize / chunksize;
+        off_t expected_size = (off_t)nblocks * (off_t)nchunks *
+            (off_t)chunksize * (off_t)total_ranks;
+
+        /* get stat data for the file */
         errno = 0;
         struct stat sbuf;
         int stat_rc = stat(targetfile, &sbuf);
         if (stat_rc == 0) {
-            test_print(rank, "stat(%s) says filesize = %llu",
-                targetfile, (unsigned long long)sbuf.st_size);
+            /* check that stat size matches expected size */
+            if (sbuf.st_size != expected_size) {
+                test_print(rank, "%s size incorrect got %llu, expected %llu",
+                    targetfile, (unsigned long long)sbuf.st_size,
+                    (unsigned long long)expected_size);
+                ret = 1;
+            }
         } else {
-            test_print(rank, "stat(%s) failed: errno=%d (%s)",
+            /* our call to stat failed */
+            test_print(rank, "stat(%s) failed:",
                 targetfile, errno, strerror(errno));
```

Comment:
@adammoody, looks like you might have dropped some format tags on line 371. There's one tag but three arguments on line 372.

Reply:
yeah the "failed:" portion is missing an input

Reply:
Nice catch!  Forgot to drop those arguments.  Turns out that test_print adds them if errno is non-zero, so dropped the formatting flags but forgot the args.

Reply:
That works too!